src/resmom/requests.c

```diff
@@ -746,16 +750,30 @@ message_job(job *pjob, enum job_file jft, char *text)
 	else if (len == 1)
 		jft = StdOut;	/* only have stdout open */
 
+#ifdef WIN32
 	if ((fds = open_std_file(pjob, jft, O_WRONLY|O_APPEND,
 		pjob->ji_qs.ji_un.ji_momt.ji_exgid)) < 0)
 		return PBSE_MOMREJECT;
 
-#ifdef WIN32
 	/* set to append mode */
 	SetFilePointer((HANDLE)_get_osfhandle(fds), (LONG)NULL,
 		(PLONG)NULL, FILE_END);
-#endif
+#else
+	while ((slept < 5) && (fds < 0)) {
+		fds = open_std_file(pjob, jft, O_WRONLY | O_APPEND | O_NONBLOCK,
+	                         pjob->ji_qs.ji_un.ji_momt.ji_exgid);
+		if (fds < 0) {
+			if ( errno == EAGAIN || errno == EWOULDBLOCK) {
```

Comment:
seteuid() and seteguid() do not set errno to EAGAIN. However, setresuid() and setresgid() do. If any of these calls failed and set errno to EAGAIN, the attempt to open the file was not successful as open() is not called at all. We would need to retry as EAGAIN is a temporary error.
It is my view, it really does not matter which system call failed and set the errno to EAGAIN in this case.

Reply:
I disagree, we are trying to find a problem in one call and checking errno which can potentially be set by another system call.
In function open_file_as_user() open is called before calling setresuid/setresgid. This means if open fails for some reason you do not want to retry, this code might end up retrying even in that case because it will be working off of errno set by setresuid/setresgid calls.